tensordict/prototype/tensorclass.py

```diff
@@ -427,14 +428,11 @@ def _getitem(self, item: NestedKey) -> Any:
     if isinstance(item, str) or (
         isinstance(item, tuple) and all(isinstance(_item, str) for _item in item)
     ):
-        raise ValueError("Invalid indexing arguments.")
+        raise ValueError(f"Invalid indexing arguments: {item}.")
     tensor_res = self._tensordict[item]
     non_tensor_res = {}
     for key, value in self._non_tensordict.items():
-        if is_tensorclass(value):
-            non_tensor_res[key] = _getitem(value, item)
-        else:
-            non_tensor_res[key] = value
+        non_tensor_res[key] = value
```

Comment:
This is just making a shallow copy now right? Could do something like `non_tensor_res = copy(self._non_tensordict)` or `non_tensor_res = dict(self._non_tensordict)` instead